packages/graphql/src/translate/queryAST/ast/operations/ConnectionReadOperation.ts

```diff
@@ -225,12 +225,14 @@ export class ConnectionReadOperation extends Operation {
     private createProjectionMapForEdge(context: QueryASTContext<Cypher.Node>): Cypher.Map {
         const nodeProjectionMap = this.generateProjectionMapForFields(this.nodeFields, context.target);
         if (nodeProjectionMap.size === 0) {
-            const targetNodeName = this.target.name;
             nodeProjectionMap.set({
-                __resolveType: new Cypher.Literal(targetNodeName),
                 __id: Cypher.id(context.target),
             });
         }
+        const targetNodeName = this.target.name;
+        nodeProjectionMap.set({
+            __resolveType: new Cypher.Literal(targetNodeName),
+        });
```

Comment:
```suggestion
        nodeProjectionMap.set({
            __resolveType: new Cypher.Literal(this.target.name),
        });
```

Just a thought. I don't this particularly helpful like some other variables we create to help with clarity.

Reply:
fair enough